connectivity/check/policy.go

```diff
@@ -16,21 +16,126 @@ package check
 
 import (
 	"context"
-	"encoding/json"
+	"errors"
 	"fmt"
-	"regexp"
 	"strconv"
 	"strings"
 	"time"
 
 	"github.com/cilium/cilium-cli/defaults"
+	"github.com/cilium/cilium-cli/internal/k8s"
 	"github.com/cilium/cilium/pkg/k8s/client/clientset/versioned/scheme"
+
 	"k8s.io/apimachinery/pkg/runtime/serializer"
 
 	ciliumv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+// getCiliumPolicyRevisions returns the current policy revisions of all Cilium pods
+func (ct *ConnectivityTest) getCiliumPolicyRevisions(ctx context.Context) (map[Pod]int, error) {
+	revisions := make(map[Pod]int)
+	for _, cp := range ct.ciliumPods {
+		revision, err := getCiliumPolicyRevision(ctx, cp)
+		if err != nil {
+			return revisions, err
+		}
+		revisions[cp] = revision
+	}
+	return revisions, nil
+}
+
+// waitCiliumPolicyRevisions waits for the Cilium policy revisions to be bumped
+// TODO: Improve error returns here, currently not possible for the caller to reliably detect timeout.
+func (t *Test) waitCiliumPolicyRevisions(ctx context.Context, revisions map[Pod]int) error {
+	var err error
+	for pod, oldRevision := range revisions {
+		err = waitCiliumPolicyRevision(ctx, pod, oldRevision+1, defaults.PolicyWaitTimeout)
```

Comment:
You can do the random seed in a follow-up PR. I have a fix for the policy race which turned out to be ipcache race instead. `cilium policy wait` works as advertized :-)

Reply:
Hmm, you're right in that we might want to reproduce a test run with the same sequence at some point. The design of the runner should guarantee that each Test gets its own independent context with minimal to no state shared between them, which should make this deterministic execution unnecessary.

I would prefer to avoid inheriting these types of features from the ginkgo suite, but I don't feel too strongly against this one in particular. Sounds like something with the lowest priority, though.